File: Array/String/07_program.c
// password checking with strcmp()

#include <stdio.h>
#include <string.h>
#include <unistd.h>

#define MAX_INPUT 100

void countdown(int seconds){
    for(int i = seconds; i > 0; i--){
        printf("\rPlease wait %d seconds...", i);
        fflush(stdout);
        sleep(1);
    }
    printf("\r\033[2K");
    printf("You can try again now.\n");
}

int main(){

    char storedPassword[] = "password";
    char enteredPassword[MAX_INPUT];

    int maxAttempts = 3;
    int attempts = 0;

    int delaySeconds = 30;

    while(1){

        while(attempts<maxAttempts){
            printf("Enter password: ");
            
            fgets(enteredPassword, sizeof(enteredPassword), stdin);
            enteredPassword[strcspn(enteredPassword, "\n")]= '\0';
            
            if(strcmp(storedPassword, enteredPassword)==0){
                printf("Correct!\n");
                return 0;
            }else{
                printf("Incorrect! Try again!\n");
                attempts++;
            }
        }
        printf("Maximum attempts reached. You must wait %d seconds.\n", delaySeconds);
        countdown(delaySeconds);
        attempts = 0;
    }
    return 0; 

}

File: Array/String/07_program.py
import time

def main():
    stored_password = "mySecretPassword"
    max_attempts = 3
    delay_in_seconds = 30

    while True:
        for _ in range(max_attempts):
            if input("Enter the password: ") == stored_password:
                print("Password is correct!")
                return
            print("Incorrect password! Try again.")

        print(f"Maximum attempts reached. Waiting {delay_in_seconds} seconds...")
        for i in range(delay_in_seconds, 0, -1):
            print(f"\rPlease wait {i} seconds...", end='', flush=True)
            time.sleep(1)

        print("\r\033[2KYou can try again now.")

main()
